File: <100/036/036.jl
string_0b(n:: Int):: String = string(n, base=2)
is_palindrome(s:: String):: Bool = reverse(s) == s
is_ok(n:: Int):: Bool = n |> string |> is_palindrome ? n |> string_0b |> is_palindrome : false

function main()::Nothing
  upper_limit:: Int = 10^6-1
  values = (val for val ∈ 1:upper_limit if is_ok(val))
  values |> sum |> println
  
  return nothing
end

@time main()

File: <100/036/036.hs
import Numeric (showBin)


main :: IO ()
main = print $ sum [n | n <- [1..upperLimit], isBothPalindrome n]


isBothPalindrome :: Integer -> Bool
isBothPalindrome n = isP10 && isP2
    where
        isP10 = isPalindrome . show $ n
        isP2  = isPalindrome . showBin' $ n


isPalindrome :: Eq a => [a] -> Bool
isPalindrome s = reverse s == s


showBin' :: Integer -> String
showBin' = flip showBin ""


upperLimit :: Integer
upperLimit = 10^6
